zds/search/utils.py

```diff
@@ -111,8 +112,21 @@ def reindex_content(versioned, publishable_content):
     :return:
     """
 
-    # We just delete all index that correspond to the content
-    SearchIndexContent.objects.filter(publishable_content=publishable_content).delete()
+    if SearchIndexContent.objects.filter(publishable_content=publishable_content).count() > 0:
+        search_index_content_ = SearchIndexContent.objects.filter(publishable_content=publishable_content).get()
+
+        # We may need to update his pub_date before deleting it
+        if search_index_content_.pubdate is None:
+            search_index_content_.pubdate = datetime.now()
+
+        if search_index_content_.update_date is None:
+            search_index_content_.update_date = datetime.now()
+
+        if search_index_content_.pubdate is None or search_index_content_.update_date is None:
```

Comment:
` le "marchera pas. Tu viens de les mettre à jour."`

tu viens de forcer une valeur par défeau à

search_index_content_.pubdate et

search_index_content_.update_date

et là tu testes

if search_index_content_.pubdate is None or
search_index_content_.update_date is None



2015-08-21 14:29 GMT+02:00 Hugo Courtecuisse <notifications@github.com>:

> In zds/search/utils.py
> <https://github.com/artragis/zds-site/pull/277#discussion_r37629305>:
>
> > @@ -111,8 +112,21 @@ def reindex_content(versioned, publishable_content):
> >      :return:
> >      """
> >
> > -    # We just delete all index that correspond to the content
> > -    SearchIndexContent.objects.filter(publishable_content=publishable_content).delete()
> > +    if SearchIndexContent.objects.filter(publishable_content=publishable_content).count() > 0:
> > +        search_index_content_ = SearchIndexContent.objects.filter(publishable_content=publishable_content).get()
> > +
> > +        # We may need to update his pub_date before deleting it
> > +        if search_index_content_.pubdate is None:
> > +            search_index_content_.pubdate = datetime.now()
> > +
> > +        if search_index_content_.update_date is None:
> > +            search_index_content_.update_date = datetime.now()
> > +
> > +        if search_index_content_.pubdate is None or search_index_content_.update_date is None:
>
> Nan, ça ok ^^ le "marchera pas. Tu viens de les mettre à jour."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artragis/zds-site/pull/277/files#r37629305>.
>